server/src/main/java/org/apache/uniffle/server/ShuffleServerGrpcService.java

```diff
@@ -428,6 +428,20 @@ public void finishShuffle(
   public void requireBuffer(
       RequireBufferRequest request, StreamObserver<RequireBufferResponse> responseObserver) {
     String appId = request.getAppId();
+    if (shuffleServer.isActivateClientPartitionReassign()
+        && shuffleServer
+            .getShuffleTaskManager()
+            .getShuffleTaskInfo(appId)
+            .isBlockFailureReassignEnabled()) {
+      responseObserver.onNext(
+          RequireBufferResponse.newBuilder()
+              .setStatus(StatusCode.SERVER_INACTIVE.toProto())
+              .setRetMsg("Server is inactive, status: " + shuffleServer.getServerStatus())
+              .build());
+      responseObserver.onCompleted();
+      return;
+    }
+
```

Comment:
This may easily trigger the restriction of `blockFailSentRetryMaxTimes` in reassign

Reply:
Why? It's rare that server is not healthy

Reply:
In our production environment, our health check uses HealthScriptChecker. The server is not healthy very frequently, such as high load or high network card traffic or disk utilization triggering threshold, etc.

Reply:
> In our production environment, our health check uses HealthScriptChecker. The server is not healthy very frequently, such as high load or high network card traffic or disk utilization triggering threshold, etc.

Yes, I introduced the extra config to control whether to enable this feature. So you could disable this.